packages/@lwc/engine-core/src/framework/api.ts

```diff
@@ -780,6 +787,12 @@ function dc(
     if (Ctor == null) {
         return null;
     }
+    let props = {};
+    if (isLightningElementConfig(Ctor)) {
+        props = Ctor.props;
+        Ctor = Ctor.constructor;
```

Comment:
Using the actual string `'constructor'` on a plain object can have weird side effects. E.g. it'll be defined even if the object is empty:

```
> const obj = {}
undefined
> obj.constructor
[Function: Object]
```

  I would suggest renaming it to `ctor` or something.